instruments/teledyne/maui.py

```diff
@@ -0,0 +1,1404 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Provides support for the Teledyne-Lecroy Oscilloscopes that use the
+MAUI interface.
+
+Development follows the IEEE 488.2 Command Reference from the MAUI
+Oscilloscopes Remote Control and Automation Manual, document number
+maui-remote-control-automation_10mar20.pdf
+
+Where possible, commands are sent using the enum_property, ... that
+are usually used for SCPI classes, even though, this is not an SCPI
+class.
+"""
+
+# IMPORTS #####################################################################
+
+from enum import Enum
+import numpy as np
+
+from instruments.abstract_instruments import (
+    Oscilloscope, OscilloscopeChannel, OscilloscopeDataSource
+)
+import instruments.units as u
+from instruments.util_fns import (
+    assume_units, enum_property, bool_property, ProxyList
+)
+
+# CLASSES #####################################################################
+
+
+# pylint: disable=too-many-lines,arguments-differ
+
+
+class MAUI(Oscilloscope):
```

Comment:
My initial reaction was to rename this `Maui` instead of `MAUI`, thoughts?

Reply:
Teledyne-LeCroy in fact uses MAUI as an abbreviation [(see here)](https://teledynelecroy.com/maui/): Most Advanced User Interface... 
In PEP8, [this section](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles), it is stated for CamelCase:

 > Note: When using acronyms in CapWords, capitalize all the letters of the acronym. Thus HTTPServerError is better than HttpServerError.

I honestly don't have strong feelings either way and am perfectly fine to rename it. Let me know what you think.

Reply:
AH, okay. We'll keep it what it is then. :+1: